crates/sargon/src/profile/logic/profile_network/profile_network_entities_linked_to_factor_source.rs

```diff
@@ -0,0 +1,111 @@
+use crate::prelude::*;
+
+impl ProfileNetwork {
+    /// Returns the entities linked to a given `FactorSource` on the current `ProfileNetwork`.
+    pub fn entities_linked_to_factor_source(
+        &self,
+        factor_source: FactorSource,
+        integrity: FactorSourceIntegrity,
+    ) -> Result<EntitiesLinkedToFactorSource> {
+        let accounts = self
+            .accounts_non_hidden()
+            .iter()
+            .filter(|a| {
+                a.security_state
+                    .is_linked_to_factor_source(factor_source.clone())
```

Comment:
This could be in a common function, then passed to filter.

Reply:
that what was my original idea: to define the `filter` predicate once and then use it on every collection:
```Rust
let filter = |e: &dyn HasSecurityState| {
    e.security_state().is_linked_to_factor_source(factor_source.clone())
};
```

I wasn't able to make it work with a closure, but using a function that sends the `factor_source` each time works, thanks!